devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java

```diff
@@ -0,0 +1,285 @@
+
+        if (LaunchMode.TEST.equals(mode)) {
+            scannedConfigurations.add(JavaPlugin.TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME);
+        } else {
+            scannedConfigurations.add(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME);
+        }
+
+        if (LaunchMode.DEVELOPMENT.equals(mode)) {
+            scannedConfigurations.add(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME);
+        }
+
+        final Collection<org.gradle.api.artifacts.Dependency> directExtensionDependencies = getDirectExtension(project);
+
+        final Set<Dependency> appDependencies = new HashSet<>();
+        for (String configurationName : scannedConfigurations) {
+            final ResolvedConfiguration configuration = project.getConfigurations().getByName(configurationName)
+                    .getResolvedConfiguration();
+            appDependencies.addAll(collectDependencies(configuration, configurationName, mode, project));
+            directExtensionDependencies
+                    .addAll(getDirectExtensionDependencies(configuration.getFirstLevelModuleDependencies(), appDependencies));
+        }
+
+        final Set<Dependency> extensionDependencies = collectExtensionDependencies(project, directExtensionDependencies);
+
+        ArtifactCoords appArtifactCoords = new ArtifactCoordsImpl(project.getGroup().toString(), project.getName(),
+                project.getVersion().toString());
+
+        return new QuarkusModelImpl(getWorkspace(project),
+                getAdditionalWorkspaces(project.getRootProject(), project.getName()),
+                appDependencies,
+                extensionDependencies);
+    }
+
+    private Workspace getWorkspace(Project project) {
+        ArtifactCoords appArtifactCoords = new ArtifactCoordsImpl(project.getGroup().toString(), project.getName(),
+                project.getVersion().toString());
+        final SourceSet mainSourceSet = QuarkusGradleUtils.getSourceSet(project, SourceSet.MAIN_SOURCE_SET_NAME);
+
+        return new WorkspaceImpl(appArtifactCoords, project.getProjectDir().getAbsoluteFile(),
+                project.getBuildDir().getAbsoluteFile(), getSourceSourceSet(mainSourceSet), convert(mainSourceSet));
+    }
+
+    private Set<Workspace> getAdditionalWorkspaces(Project project, String mainProjectName) {
+        Set<Workspace> workspaces = new HashSet<>();
+        for (Project subproject : project.getSubprojects()) {
+
+            if (subproject.getName().equals(mainProjectName)) {
+                continue;
+            }
+            final Convention convention = subproject.getConvention();
+            JavaPluginConvention javaConvention = convention.findPlugin(JavaPluginConvention.class);
+            if (javaConvention == null) {
+                continue;
+            }
+
+            workspaces.add(getWorkspace(subproject));
+        }
+        return workspaces;
+    }
+
+    private Set<org.gradle.api.artifacts.Dependency> getDirectExtension(Project project) {
```

Comment:
This method should be renamed. It's not about extensions here. All it does it collects the enforced platforms. It should reflect that in the name.

Reply:
👍 , I will rename it